runtime/src/accounts.rs

```diff
@@ -299,6 +300,14 @@ impl Accounts {
                                 )
                             })
                             .collect::<Result<TransactionLoaders>>()?;
+
+                        // Load sysvar accounts that will be made available to programs via syscalls
```

Comment:
@sakridge Is there a better way to load the sysvar accounts in order to make them available to programs via a syacall API?

Reply:
Incurring this cost for every single transaction doesn't feel grate.   Seems like we need a new on-demand interface to load read-only accounts from within the syscall handler

Reply:
Yeah, not sure which is better, grabbing the lock whenever someone asks for a sysvar or loading the sysvar account as readonly from cache while the lock is already held...

Reply:
If we put `bank.ancestors` behind `Rc` we could share the bank's accounts and ancestors with invoke_context to load the sysvars on demand, we could load on the initial request in the instruction.  @sakridge any thoughts on this and general direction?

Reply:
Loading on-demand is preferred I think and `bank.ancestors` in `Rc` also sounds reasonable.

Reply:
Another option is we create a distinct external call for each sysvar which will result in unresolved external symbols for each syscall the program calls.  With that, we know what syscalls the program is planning to call and we can load just those once as a bundle with a single lock.  If some major restructuring, we could make that information known during the initial account load and do all the loading in the initial load lock.

Reply:
> create a distinct external call for each sysvar

That's a cool idea.  One downside I see is that this'll cause sysvar  account loads for all instructions, even if the sysvar is never used

Reply:
It does sound attractive to have them be able to be scanned in the program and to be able to load them in this way. I'm thinking about a GPU implementation and like a thunk-style implementation will be kind of hard to do. It would be better to know it upfront and load them before getting into the interpreter itself. This scan could be done on program use and cached somewhere as well similar to the JIT binary.

Reply:
@mvines Are you referring to the case where the transaction fails before a particular syscall is called?

Reply:
Just because there's a unresolved sysvar symbol doesn't mean that all program instructions will require that sysvar, maybe the sysvar is only used one in a million instructions.

Although I hear sakridge's point about planning for future GPU support, where loading then all once up front to avoid a GPU stall is probably better than optimizing the low TPS case

Reply:
I'm planning on loading on demand for now and exporting separate symbols for each sysvar so that we have the flexibility to optimize in the future.  Any objections to that direction?

Reply:
sgtm, on-demand loading for cpu case seems great, with the option to batch if/when there's GPU offload available

Reply:
sounds good